modules/models/record.php

```diff
@@ -138,20 +149,14 @@ public function delete($pk="id") // Deletes by assuming $id is always the pk. No
             throw new ErrorException("Primary key specified is not valid as it returns more than one record. PK: ".$pk);
         }
 
-        if (is_numeric($pk) and !is_string($pk)) {
-            $condition = sprintf("%s = %s", $pk, $this->$pk);
-        } else {
-            $condition = sprintf("%s = '%s'", $pk, $this->$pk);
-        }
-
         $sql = sprintf(
-            "DELETE FROM %s WHERE %s",
+            "DELETE FROM %s WHERE %s = :%s",
             get_called_class()::$table_name,
-            $condition
+            $pk,
+            $pk
         );
```

Comment:
if im not mistaken, using the colon allows us to take advantage of PDO's sql sanitisation (I havent verified this, its just a guess)